pkg/providers/amifamily/resolver.go

```diff
@@ -113,29 +114,44 @@ func (r Resolver) Resolve(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTem
 	}
 	var resolvedTemplates []*LaunchTemplate
 	for amiID, instanceTypes := range amiIDs {
-		resolved := &LaunchTemplate{
-			Options: options,
-			UserData: amiFamily.UserData(
-				r.defaultClusterDNS(options, machine.Spec.Kubelet),
-				append(machine.Spec.Taints, machine.Spec.StartupTaints...),
-				options.Labels,
-				options.CABundle,
-				instanceTypes,
-				nodeTemplate.Spec.UserData,
-			),
-			BlockDeviceMappings: nodeTemplate.Spec.BlockDeviceMappings,
-			MetadataOptions:     nodeTemplate.Spec.MetadataOptions,
-			DetailedMonitoring:  aws.BoolValue(nodeTemplate.Spec.DetailedMonitoring),
-			AMIID:               amiID,
-			InstanceTypes:       instanceTypes,
+		maxPodsToInstanceTypes := map[int][]*cloudprovider.InstanceType{}
+		for _, instanceType := range instanceTypes {
+			maxPodsToInstanceTypes[int(instanceType.Capacity.Pods().Value())] = append(maxPodsToInstanceTypes[int(instanceType.Capacity.Pods().Value())], instanceType)
 		}
-		if resolved.BlockDeviceMappings == nil {
-			resolved.BlockDeviceMappings = amiFamily.DefaultBlockDeviceMappings()
+		for maxPods, instanceTypes := range maxPodsToInstanceTypes {
```

Comment:
Do we want to add comments here to describe why we are doing this logic since it's not immediately intuitive to me reading it for the first time?